packages/video_player/video_player/lib/video_player.dart

```diff
@@ -1089,58 +1084,52 @@ class VideoProgressIndicator extends StatefulWidget {
 }
 
 class _VideoProgressIndicatorState extends State<VideoProgressIndicator> {
-  _VideoProgressIndicatorState() {
-    listener = () {
-      if (!mounted) {
-        return;
-      }
-      setState(() {});
-    };
-  }
-
-  late VoidCallback listener;
-
   VideoPlayerController get controller => widget.controller;
 
   VideoProgressColors get colors => widget.colors;
 
+  void _didUpdateControllerValue() {
+    setState(() {
+      // The build method reads from controller.value.
```

Comment:
The `mounted` check wasn't necessary. The state will stop listening when the element is unmounted.